vllm/v1/core/scheduler.py

```diff
@@ -73,8 +76,9 @@ def __init__(
 
         # OPTIMIZATION: Cache the CachedRequestData objects to avoid creating
         # them at each scheduling step.
-        # Request id -> CachedRequestData
-        self._cached_reqs_data: Dict[str, CachedRequestData] = {}
+        # Request id -> Queue of CachedRequestData
+        self._cached_reqs_data: Dict[
+            str, Queue[CachedRequestData]] = defaultdict(Queue)
```

Comment:
Why is it a `Queue` rather than a `List`?

Reply:
Queue is faster at removing and appending objects